lib/wasi/src/syscalls/legacy/snapshot0.rs

```diff
@@ -22,17 +23,18 @@ pub fn fd_filestat_get(
     let new_buf: WasmPtr<types::__wasi_filestat_t> = buf.cast();
 
     // Copy the data including the extra data
-    let new_filestat_setup: types::__wasi_filestat_t = wasi_try_mem!(new_buf.read(memory));
+    let new_filestat_setup: types::__wasi_filestat_t = wasi_try_mem!(new_buf.read(&ctx, memory));
```

Comment:
Does this compile? From what I read on the `WasmPtr` wasmer3 API the signature (https://github.com/wasmerio/wasmer/blob/wasmer3/lib/api/src/sys/ptr.rs#L186) only requires the memory reference.
Am I missing something?

Reply:
It does build, and `make test-wasi` on linux x86_64 also works.

![image](https://user-images.githubusercontent.com/3250792/170317571-a0e4d047-b2a3-44ec-9573-151f477ce4d1.png)

From what I have seen, all functions require the Context.